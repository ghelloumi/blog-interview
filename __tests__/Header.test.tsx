import React from 'react';
import { render } from '@testing-library/react';
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import '@testing-library/jest-dom/extend-expect';

import Header from '../components/Organisms/Header';

describe('Enzyme tests', () => {
    test('should render Header according to snapshot', () => {
        const wrapper = mount(<Header />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})

describe('React testing libray tests', () => {
    test('should render Header with title and 2 links', () => {
        const { getAllByTestId, getByTestId } = render(<Header />);
        const title = getByTestId('title');
        expect(title.textContent.trim()).toBe('Coding test with Nextjs')
        const navlink = getAllByTestId('navlink');
        expect(navlink.length).toBe(2)
    })
})
