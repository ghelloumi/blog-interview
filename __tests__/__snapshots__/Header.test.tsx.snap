// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enzyme tests should render Header according to snapshot 1`] = `
<Header>
  <Header__Wrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Header__Wrapper-sc-1hago8x-0",
            "isStatic": true,
            "lastClassName": "cBQqfg",
            "rules": Array [
              "padding:1rem 0 3rem;",
            ],
          },
          "displayName": "Header__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Header__Wrapper-sc-1hago8x-0",
          "target": "header",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <header
        className="Header__Wrapper-sc-1hago8x-0 cBQqfg"
      >
        <Header__Title
          data-testid="title"
        >
          <StyledComponent
            data-testid="title"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Header__Title-sc-1hago8x-1",
                  "isStatic": true,
                  "lastClassName": "flkuHf",
                  "rules": Array [
                    "color:#000000;font-size:3rem;text-decoration:none;",
                  ],
                },
                "displayName": "Header__Title",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Header__Title-sc-1hago8x-1",
                "target": "h1",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h1
              className="Header__Title-sc-1hago8x-1 flkuHf"
              data-testid="title"
            >
              Coding test with Nextjs
            </h1>
          </StyledComponent>
        </Header__Title>
        <nav>
          <Header__NavList>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "Header__NavList-sc-1hago8x-2",
                    "isStatic": true,
                    "lastClassName": "esosuA",
                    "rules": Array [
                      "display:flex;list-style-type:none;margin:0;",
                    ],
                  },
                  "displayName": "Header__NavList",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "Header__NavList-sc-1hago8x-2",
                  "target": "ul",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <ul
                className="Header__NavList-sc-1hago8x-2 esosuA"
              >
                <li
                  data-testid="navlink"
                >
                  <Link
                    href="/"
                    prefetch={true}
                  >
                    <Header__NavItem
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "Header__NavItem-sc-1hago8x-3",
                              "isStatic": true,
                              "lastClassName": "zqvkG",
                              "rules": Array [
                                "color:#999;font-size:0.9rem;margin-right:1.3rem;text-decoration:none;cursor:pointer;&:hover{color:#666;}&:active{color:#333;}",
                              ],
                            },
                            "displayName": "Header__NavItem",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "Header__NavItem-sc-1hago8x-3",
                            "target": "a",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={[Function]}
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                      >
                        <a
                          className="Header__NavItem-sc-1hago8x-3 zqvkG"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                        >
                          Home
                        </a>
                      </StyledComponent>
                    </Header__NavItem>
                  </Link>
                </li>
                <li
                  data-testid="navlink"
                >
                  <Link
                    href="/blog"
                    prefetch={true}
                  >
                    <Header__NavItem
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "Header__NavItem-sc-1hago8x-3",
                              "isStatic": true,
                              "lastClassName": "zqvkG",
                              "rules": Array [
                                "color:#999;font-size:0.9rem;margin-right:1.3rem;text-decoration:none;cursor:pointer;&:hover{color:#666;}&:active{color:#333;}",
                              ],
                            },
                            "displayName": "Header__NavItem",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "Header__NavItem-sc-1hago8x-3",
                            "target": "a",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={[Function]}
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                      >
                        <a
                          className="Header__NavItem-sc-1hago8x-3 zqvkG"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                        >
                          Blog
                        </a>
                      </StyledComponent>
                    </Header__NavItem>
                  </Link>
                </li>
                <li
                  data-testid="navlink"
                >
                  <Link
                    href="/about"
                    prefetch={true}
                  >
                    <Header__NavItem
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "Header__NavItem-sc-1hago8x-3",
                              "isStatic": true,
                              "lastClassName": "zqvkG",
                              "rules": Array [
                                "color:#999;font-size:0.9rem;margin-right:1.3rem;text-decoration:none;cursor:pointer;&:hover{color:#666;}&:active{color:#333;}",
                              ],
                            },
                            "displayName": "Header__NavItem",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "Header__NavItem-sc-1hago8x-3",
                            "target": "a",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={[Function]}
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                      >
                        <a
                          className="Header__NavItem-sc-1hago8x-3 zqvkG"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                        >
                          About
                        </a>
                      </StyledComponent>
                    </Header__NavItem>
                  </Link>
                </li>
              </ul>
            </StyledComponent>
          </Header__NavList>
        </nav>
      </header>
    </StyledComponent>
  </Header__Wrapper>
</Header>
`;
